lib/common/ns.c

```diff
@@ -141,6 +167,28 @@ static void exchange_tree_edges(edge_t * e, edge_t * f)
     ND_tree_in(n).list[ND_tree_in(n).size] = NULL;
 }
 
+// init_rank() does an inital ranking to start things off.
+//
+// QUESTION: Where and what are rank values set to before init_rank()?
+//           This is important because it uses the current values of ranks
+//           to initialize the ranks.  It seems like it expects them to be set to
+//           zero or some baseline to begin with, but I have not found where
+//           that is done.
```

Comment:
I am not sure I understand this question. You mean the `ND_rank(…)` values? Memory allocated with `gv_alloc`, `gv_calloc`, or `calloc` is zeroed on allocation. So a node allocated with one of these will end up with an initial rank 0. Is that what you were asking?